File: sty_algorithm/src/main/java/com/sty/algorithm/leetcode/editor/cn/[1]两数之和.java
package com.sty.algorithm.leetcode.editor.cn;

import java.util.HashMap;
import java.util.Map;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution1 {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        int[] res = new int[2];

        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(target - nums[i])) {
                res[0] = map.get(target - nums[i]);
                res[1] = i;
                break;
            }
            map.put(nums[i], i);
        }

        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: sty_algorithm/src/main/scala/com/sty/algorithm/leetcode/editor/cn/[1]两数之和.scala
//leetcode submit region begin(Prohibit modification and deletion)
import scala.collection.mutable
object Solution1 {
    // 导入包
    import scala.collection.mutable

    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        // key存储值，value存储下标
        val map = new mutable.HashMap[Int, Int]()
        for (i <- nums.indices) {
            val tmp = target - nums(i) // 计算差值
            // 如果这个差值存在于map，则说明找到了结果
            if (map.contains(tmp)) {
                return Array(map(tmp), i)
            }
            // 如果不包含把当前值与其下标放到map
            map.put(nums(i), i)
        }
        // 如果没有找到直接返回一个空的数组，return关键字可以省略
        new Array[Int](2)
    }
}
//leetcode submit region end(Prohibit modification and deletion)
